include/openssl/service_indicator.h

```diff
@@ -0,0 +1,40 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef AWSLC_HEADER_SERVICE_INDICATOR_H
+#define AWSLC_HEADER_SERVICE_INDICATOR_H
+
+#if defined(__cplusplus)
+extern "C" {
+#endif
+
+OPENSSL_EXPORT uint64_t awslc_fips_service_indicator_get_counter(void);
+OPENSSL_EXPORT uint32_t awslc_fips_service_indicator_get_serviceID(void);
+
+#if defined(AWSLC_FIPS)
+#define IS_FIPS_APPROVED_CALL_SERVICE(approved, func)                       \
+  do {                                                                      \
+    (approved) = 0;                                                         \
+    uint64_t counter = awslc_fips_service_indicator_get_counter();          \
+    func;                                                                   \
+    if (awslc_fips_service_indicator_get_counter() != counter) {            \
+        (approved) = 1;                                                     \
+    }                                                                       \
+ }                                                                          \
+ while(0)                                                                   \
+
+#else
+#define IS_FIPS_APPROVED_CALL_SERVICE(approved, func, ...)                  \
+  do {                                                                      \
+    (approved) = 0;                                                         \
```

Comment:
You can argue what to return here... Consuming code would likely look for `approved != 0` or something (should we have a macro here, to lock into a value?).

So, for non-fips, this would fail all the time.